frmts/heif/heifdataset.cpp

```diff
@@ -748,7 +748,7 @@ CPLErr GDALHEIFRasterBand::IReadBlock(int nBlockXOff, int nBlockYOff,
             }
         }
     }
-
+    heif_image_release(hImage);
```

Comment:
shouldn't that be done too in the error code path at line 719 ?

Reply:
I don't _think_ so. That is the error path for where we are filling in `hImage` and it will be untouched on failure.

(Reason for emphasis - I'm on low sleep, so maybe missing something here).

Reply:
you're right. Just checked that https://github.com/strukturag/libheif/blob/bee0699e0e4eca18325d4d886d0590990ad51443/libheif/api/libheif/heif.cc#L1481 only sets a not-null hImage in the success code path, so no need to do heif_image_release()